lib/httpoison/base.ex

```diff
@@ -160,11 +160,11 @@ defmodule HTTPoison.Base do
       def request(method, url, body \\ "", headers \\ [], options \\ []) do
         options = process_request_options(options)
         url =
-        if Keyword.has_key?(options, :params) do
-          url <> "?" <> URI.encode_query(options[:params])
-        else
-          url
-        end
+          cond do
```

Comment:
So there are now 3 cases here. I suspect we should have now 3 tests (1 for each case) right? How can we reach the `true` case?